gallery_dl/extractor/tiktok.py

```diff
@@ -0,0 +1,207 @@
+POST_PATTERN = USER_PATTERN + \
+    r"/+(?:[pP][hH][oO][tT][oO]|[vV][iI][dD][eE][oO])/+(?:[0-9]+)/*"
+VM_POST_PATTERN = r"(?:(?:https?://)?(?:(?:vm|vt)\.)?tiktok\.com/+.*/*)|" + \
+    r"(?:(?:https?://)?(?:www\.)?tiktok\.com/+t/+.*/*)"
+SHARE_PATTERN = BASE_PATTERN + r"/+share/+video/+(?:[0-9]+)/*"
+
+
+class TiktokExtractor(Extractor):
+    """Base class for TikTok extractors"""
+
+    category = "tiktok"
+    directory_fmt = ("{category}", "{user}")
+    filename_fmt = "{title} [{id}{index:?_//}{img_id:?_//}].{extension}"
+    archive_fmt = "{id}_{index}_{img_id}"
+    root = "https://www.tiktok.com/"
+    cookies_domain = ".tiktok.com"
+
+    def urls(self):
+        return [self.url]
+
+    def avatar(self):
+        return False
+
+    def items(self):
+        videos = self.config("videos", True)
+        # We assume that all of the URLs served by urls() come from the same
+        # author.
+        downloaded_avatar = not self.avatar()
+        for tiktok_url in self.urls():
+            # If we can recognise that this is a /photo/ link, preemptively
+            # replace it with /video/ to prevent a needless second request.
+            # See below.
+            tiktok_url_to_use = compile(
+                escape("/photo/"),
+                IGNORECASE
+            ).sub("/video/", tiktok_url)
+            video_detail = util.json_loads(text.extr(
+                self.request(tiktok_url_to_use).text,
+                '<script id="__UNIVERSAL_DATA_FOR_REHYDRATION__" '
+                'type="application/json">',
+                '</script>'
+            ))["__DEFAULT_SCOPE__"]
+            if "webapp.video-detail" not in video_detail:
+                # Only /video/ links result in the video-detail dict we need.
+                # Try again using that form of link.
+                tiktok_url_to_use = video_detail["seo.abtest"]["canonical"] \
+                    .replace("/photo/", "/video/")
+                video_detail = util.json_loads(text.extr(
+                    self.request(tiktok_url_to_use).text,
+                    '<script id="__UNIVERSAL_DATA_FOR_REHYDRATION__" '
+                    'type="application/json">',
+                    '</script>'
+                ))["__DEFAULT_SCOPE__"]
+            video_detail = video_detail["webapp.video-detail"]
+            if "statusCode" in video_detail:
+                if video_detail["statusCode"] == 10222:
+                    raise exception.AuthorizationError(
+                        tiktok_url + ": Login required to access this post"
+                    )
+                elif video_detail["statusCode"] == 10204:
+                    raise exception.NotFoundError(tiktok_url)
+                elif video_detail["statusCode"] == 10231:
+                    raise exception.ExtractionError(
+                        tiktok_url + " is region locked, try downloading with "
+                        "a VPN/proxy connection"
+                    )
```

Comment:
```suggestion
                    raise exception.StopExtraction(
                        "%s is region locked, try downloading with "
                        "a VPN/proxy connection", tiktok_url,
                    )
```

You should generally raise StopExtraction instead of ExtractionError